debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/Wal2JsonReplicationMessage.java

```diff
@@ -139,105 +149,208 @@ public String getName() {
      * @return the value; may be null
      */
     public Object getValue(String columnName, String columnType, Value rawValue, final PgConnectionSupplier connection) {
-        switch (columnType) {
+        // we support wal2json both before & after commit 0255f2ac, which means we may get old-style type names 
+        // (eg. int2, timetz, varchar, numeric, _int4)
+        // or new-style/verbose ones
+        // (eg. smallint; time with time zone; character varying (255); decimal (10,3), integer[])
+        
+        Matcher m = TYPE_PATTERN.matcher(columnType);
+        if (!m.matches()) {
+            LOGGER.error("Failed to parse columnType for {} '{}'", columnName, columnType);
+            throw new ConnectException(String.format("Failed to parse columnType '%s' for column %s", columnType, columnName));
+        }
+        String fullType = m.group("full");
+        String baseType = m.group("base").trim();
+        if (!Objects.toString(m.group("suffix"), "").isEmpty()) {
+            baseType = String.join(" ", baseType, m.group("suffix").trim());
```

Comment:
this path isn't covered by the tests, I think a field using [`timestamp (0) with time zone`](https://www.postgresql.org/docs/9.6/static/datatype-datetime.html) should fix it. How to add?

Reply:
You can extend the table here - https://github.com/debezium/debezium/blob/master/debezium-connector-postgres/src/test/resources/postgres_create_tables.ddl#L12